hack/charts/cluster-api-operator/templates/deployment.yaml

```diff
@@ -100,9 +97,15 @@ spec:
         - containerPort: 9443
           name: webhook-server
           protocol: TCP
-        - containerPort: {{ ( split ":" $.Values.metricsBindAddr)._1 | int }}
+          {{- if $.Values.diagnosticsAddress }}
+          {{- $diagnosticsPort := $.Values.diagnosticsAddress }}
+          {{- if contains ":" $diagnosticsPort -}}
+          {{ $diagnosticsPort = ( split ":" $.Values.diagnosticsAddress)._1 }}
```

Comment:
```suggestion
          {{ $diagnosticsPort = ( split ":" $.Values.diagnosticsAddress)._1 | int }}
```

Reply:
Thanks, updated now

Reply:
the int conversion happens in 105 now to also convert not split input strings

Reply:
true, I overlooked it. Perhaps we can get rid of the int conversion on line 105?

Reply:
the now double int conversion should not break anything as well though ;)

Reply:
making double sure :)